packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -109,11 +111,17 @@ export class GadgetConnection {
     this.endpoint = options.endpoint;
     if (options.fetchImplementation) {
       this._fetchImplementation = options.fetchImplementation;
-    } else if (typeof window != "undefined" && window.fetch) {
-      this._fetchImplementation = window.fetch.bind(window);
+    } else if (typeof objectForGlobals != "undefined" && objectForGlobals.fetch) {
+      this._fetchImplementation = objectForGlobals.fetch.bind(objectForGlobals);
     } else {
       this._fetchImplementation = async (...args: [any]) => {
-        const { fetch } = await import("cross-fetch");
+        // lazily import cross-fetch to avoid bundling it in the client
+        let fetch: typeof globalThis.fetch = (await import("cross-fetch")) as any;
+
+        // when compiled to CJS, the dynamic import above returns the default export right away. in ESM though, we need to access it ourselves
+        if ((fetch as any).default) {
+          fetch = (fetch as any).default;
```

Comment:
we were previously using essentially `fetch.fetch`; is the idea that `fetch.default` will be present in both contexts and be the right thing? or is the difference here moving to cross-fetch 4?

Reply:
Good question, and to be completely honest, I don't super understand. If I run this script (raw JS, using real ESM syntax):

```javascript
import fetch from "cross-fetch";
console.log(fetch)
```

I get this:
```
❯ node packages/api-client-core/spec/imports/test.mjs
<ref *1> [Function: fetch] {
  ponyfill: true,
  fetch: [Circular *1],
  Headers: [class Headers],
  Request: [class Request],
  Response: [class Response],
  default: [Circular *1]
}
```

the imported value has a `fetch` prop *and* a `default` prop. That's true for cross-fetch v3 and v4. But, in the discord user's `next.js` setup, the imported value obviously doesn't have a `fetch` key, and just had this default key. So, I am unsure what piece of the build chain is different or not, but, I know that the default prop will point to the right thing if it does exist, so I think it is safe to just always use it.

Reply:
:+1: